lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2569,141 +2580,213 @@ buildAndSignTransactionNew
        , IsOwned s k ktype
        , IsOurs s RewardAccount
        , GenChange s
+       , AddressBookIso s
        )
-    => Tracer IO WalletLog
-    -> TimeInterpreter (Either PastHorizonException)
+    => TimeInterpreter (ExceptT PastHorizonException IO)
     -> DBLayer IO s k
     -> NetworkLayer IO Block
     -> TransactionLayer k ktype SealedTx
+    -> Passphrase "user"
     -> WalletId
     -> ArgGenChange s
     -> AnyRecentEra
+    -> PreSelection
+    -> TransactionCtx
+    -> IO (BuiltTx, UTCTime)
+buildAndSignTransactionNew ti db@DBLayer{..} netLayer txLayer pwd walletId
+    argGenChange era preSelection txCtx = do
+    --
+    stdGen <- initStdGen
+    pureTimeInterpreter <- snapshot ti
+    protocolParameters <- currentProtocolParameters netLayer
+    let wrapError = ExceptionWitnessTx . ErrWitnessTxWithRootKey
+    throwOnError <=< runExceptT $ withRootKey db walletId pwd wrapError $
+        \rootKey scheme -> lift $ atomically $ do
+            pendingTxs <- fmap fromTransactionInfo <$>
+                readTransactions
+                    walletId Nothing Descending wholeRange (Just Pending)
+            txBuiltForSlot <- modifyDBMaybe walletsDB $ do
+                let err = ExceptionConstructTx . ErrConstructTxNoSuchWallet
+                adjustNoSuchWallet walletId err $ \(s :: WalletState s) -> do
+                    let wallet = WalletState.getLatest s
+                    let (errorOrBuiltTx, wallet') =
+                            buildAndSignTransactionPure @k @ktype @s @n
+                                nullTracer
+                                pureTimeInterpreter
+                                (Set.fromList pendingTxs)
+                                rootKey
+                                scheme
+                                pwd
+                                protocolParameters
+                                txLayer
+                                argGenChange
+                                era
+                                preSelection
+                                txCtx
+                            & runExceptT . withExceptT
+                                (either ExceptionBalanceTx ExceptionConstructTx)
+                            & (`runStateT` wallet)
+                            & (`evalRand` stdGen)
+
+                    -- Newly generated change addresses only change the Prologue
+                    Right ( [ReplacePrologue $ getPrologue $ getState wallet']
+                          , (errorOrBuiltTx, currentTip wallet ^. #slotNo)
+                          )
+
+            let slotTime slot = liftIO $ interpretQuery
+                    (neverFails "slot is ahead of the node tip" ti)
+                    (slotToUTCTime slot)
+
+            bitraverse throwOnError slotTime =<< throwOnError txBuiltForSlot
+  where
+    throwOnError :: (MonadIO m, Exception e) => Either e a -> m a
+    throwOnError = either (liftIO . throwIO) pure
+
+data BuiltTx = BuiltTx
```

Comment:
You're right: this type is on the brink of disappearance :) 
My rule of thumb is to prefer named products over anonymous ones, 
as there is an opportunity to make code more self-documenting by giving the type
a meaningful name. 

This one in particular might disappear very soon, idk, it might survive bit longer, 
and I prefer it to carry its name proudly "I am a Tx that has just been built!" :)

Reply:
`(SealedTx, TxMeta)` is not terrible but the reason why they go together is not obvious, it feels a bit ad-hoc. I'd still name it something like `SubmittedTx`.